src/components/OptionsBar/Options/FilterModal/Filters/RelativeFilters.jsx

```diff
@@ -0,0 +1,48 @@
+import { useSelector } from 'react-redux';
+import { useRef } from 'react';
+
+const relativeFilters = (filterIds, tabIndex) => {
+  const filters = useSelector((state) => state.filters);
+
+  const getFilterMatrix = () => {
+    const filterMatrix = new Array(filterIds.length);
+
+    filterIds.forEach((id, index) => {
+      filterMatrix[index] = filters[id];
+    });
+
+    return filterMatrix;
+  };
+
+  const queuedFilters = useRef(getFilterMatrix());
+
+  const getFilters = () => queuedFilters.current;
+
+  const getRelativeFilters = () => getFilters()[tabIndex];
+
+  const pushRelativeFilters = (filter) => getRelativeFilters().push(filter);
+
+  const spliceRelativeFiltersAt = (index) => getRelativeFilters().splice(index, 1);
+
+  const rebuildFilters = () => {
+    getFilters().forEach((queuedFilter, index) => {
```

Comment:
What are you trying to do here? Force each element to have a new reference but maintain the content? Why?

Reply:
This was needed because I was working directly with the value from the store, which is immutable, so I was getting an error whenever I tried to change it. I fixed this by reworking the relative filter functions to not change the store value.